org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/ElementOrComponentTypeComputer.java

```diff
@@ -90,6 +91,14 @@ protected LightweightTypeReference doVisitUnboundTypeReference(UnboundTypeRefere
 		return null;
 	}
 	
+	@Override
+	protected LightweightTypeReference doVisitCompoundTypeReference(CompoundTypeReference reference) {
+		if (reference.isArray()) {
+			return reference.getComponentType();
+		}
+		return super.doVisitCompoundTypeReference(reference);
```

Comment:
sorry, I overlooked this in my first review.
I think we shouldn't check for reference.isArray() but just delegate to reference.getComponentType() in all cases.